### PR TITLE
[WIP] ipq40xx: convert Netgear EX61X0v2 to DSA

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Convert Netgear EX61X0v2 to DSA and enable it.

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
[netgear_ex6100v2.zip](https://github.com/openwrt/openwrt/files/10507201/netgear_ex6100v2.zip)

**Comment by Djfe** [comment]
I'll test this patch on an EX6150v2 tomorrow or on Monday

**Comment by Schneehuhn** [comment]
This patch applies to convert Netgear EX61X0v2 to DSA, i.e. both EX6100v2 and EX6150v2.

In [target/linux/ipq40xx/base-files/etc/board.d/02_network](https://github.com/openwrt/openwrt/pull/11879/commits/6f0cf8d71aee5fd99ba10461722d1001b327824e#diff-88e0ba5fe0895f3472ddedd92e479fed51c377e5c5b7575d0706104e7656cf0c), doesn't the EX6100 also need to have the LAN interface set up?

Around line 53, there should also be something like 
`netgear,ex6150v2|`

File: target/linux/ipq40xx/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -50,7 +50,8 @@ ipq40xx_setup_interfaces()
 	meraki,mr33|\
 	meraki,mr74|\
 	mikrotik,lhgg-60ad|\
-	mikrotik,sxtsq-5-ac)
+	mikrotik,sxtsq-5-ac|\
+	netgear,ex6150v2)
```
Comment (Schneehuhn) [inline target/linux/ipq40xx/base-files/etc/board.d/02_network:54]: Also needs
'netgear,ex6100v2'
to apply to the EX6100

File: target/linux/ipq40xx/image/generic.mk [hunk 2]
```diff
@@ -814,8 +814,7 @@ define Device/netgear_ex6100v2
 	DEVICE_MODEL := EX6100
 	DEVICE_VARIANT := v2
 endef
-# Missing DSA Setup
-#TARGET_DEVICES += netgear_ex6100v2
+TARGET_DEVICES += netgear_ex6100v2
```
Comment (Schneehuhn) [inline target/linux/ipq40xx/image/generic.mk:817]: A few lines further down, the EX6150 should be uncommented as well, this patch applies to both EX6100 and EX6150

**Comment by blocktrron** [comment]
This can't work. The EX61x0v2 uses a QCA8072 psgmii PHY, not a RGMII one.

**Comment by Djfe** [comment]
Very helpful comments.
Haven't had the time, yet. My first attempt before reading this PR failed though. 
I didn't add anything rgmii, yet. But I used swport5, so it's one of the ports 1-4, I guess. In my next try I was going to enable all 4 of them, so figure out which one is which.

@blocktrron have you ever connected to an ex6150v2 via serial?
I have no clue how to take the board out of the case without destroying the antennas 😅
The serial must be inside the case somewhere, is it? (there is no wiki article with pictures, yet. But I might write one since I already opened mine (I removed 4 screws and then gave up due to how it sits in the case))

**Comment by Schneehuhn** [comment]
> I have no clue how to take the board out of the case without destroying the antennas sweat_smile The serial must be inside the case somewhere, is it?

[Wikidevi](https://wikidevi.wi-cat.ru/Netgear_EX6150v2) has some photos of the EX6150v2. Have you seen that article?

**Comment by Djfe** [comment]
No, I haven't @Schneehuhn
Those work just fine for a wiki article and now from the photos I understand how to take it apart actually :)

@PolynomialDivision please close this PR, @blocktrron committed a fix last week :)
https://github.com/openwrt/openwrt/commit/bf055fcdca155993b9b6c82f951000eefb13288a

**Pull request closed without merging**